### ISSUE TITLE
generation=2 ibm_is_instance stopping state needs to be supported

### ISSUE THREAD

**Issue by powellquiring** [IBM-Cloud/terraform-provider-ibm#770]
Saw the following message during destroy.  It looks like the ibm_is_instances go through a 'stopping' state on the way to becoming stopped.  The error message should be fixed as well, notice the %!s(<nil>):

Generated the bug with this configuration, but not every time
```
variable ibmcloud_api_key {}
variable ssh_key_name {}
variable resource_group_name {}
variable generation {
  default = "1"
}



variable "ibmcloud_timeout" {
  description = "Timeout for API operations in seconds."
  default     = 900
}

variable region {
  default = "us-south"
}

variable zone {
  default = "us-south-1"
}

variable "basename" {
  description = "Name for the VPC to create and prefix to use for all other resources."
  default     = "example"
}

provider ibm {
  region           = "${var.region}"
  ibmcloud_api_key = "${var.ibmcloud_api_key}"
  ibmcloud_timeout = "${var.ibmcloud_timeout}"
  generation       = 2
}

data "ibm_resource_group" "group" {
  name = "${var.resource_group_name}"
}

resource ibm_is_vpc "vpc" {
  name           = "${var.basename}"
  resource_group = "${data.ibm_resource_group.group.id}"
}

resource ibm_is_security_group "sg1" {
  name           = "${var.basename}-sg1"
  vpc            = "${ibm_is_vpc.vpc.id}"
  resource_group = "${data.ibm_resource_group.group.id}"
}

# allow ssh access to this instance from anywhere on the planet
resource "ibm_is_security_group_rule" "ingress_ssh_all" {
  group     = "${ibm_is_security_group.sg1.id}"
  direction = "inbound"
  remote    = "0.0.0.0/0"                       # TOO OPEN for production

  tcp = {
    port_min = 22
    port_max = 22
  }
}

resource ibm_is_subnet "subnet1" {
  name                     = "${var.basename}-subnet1"
  vpc                      = "${ibm_is_vpc.vpc.id}"
  zone                     = "${var.zone}"
  total_ipv4_address_count = 256
}

data ibm_is_ssh_key "ssh_key" {
  name = "${var.ssh_key_name}"
}

data ibm_is_image "ubuntu" {
  name = "ibm-ubuntu-18-04-64"
}

locals = {
  count = 4
}

resource ibm_is_instance "vsi" {
  count = "${local.count}"
  name           = "${var.basename}-vsi${count.index}"
  vpc            = "${ibm_is_vpc.vpc.id}"
  zone           = "${var.zone}"
  keys           = ["${data.ibm_is_ssh_key.ssh_key.id}"]
  image          = "${data.ibm_is_image.ubuntu.id}"
  profile        = "c2-2x4"
  resource_group = "${data.ibm_resource_group.group.id}"

  primary_network_interface = {
    subnet          = "${ibm_is_subnet.subnet1.id}"
    security_groups = ["${ibm_is_security_group.sg1.id}"]
  }
}

resource ibm_is_floating_ip "fip1" {
  count = "${local.count}"
  name   = "${var.basename}-fip${count.index}"
  target = "${ibm_is_instance.vsi.*.primary_network_interface.0.id[count.index]}"
}
```

```
Error: Error applying plan:

1 error occurred:
	* module.vpc_pub_priv.ibm_is_instance.frontend (destroy): 1 error occurred:
	* ibm_is_instance.frontend: unexpected state 'stopping', wanted target 'stopped, failed'. last error: %!s(<nil>)
```
### Terraform Version
```
(pyvirt) ~/github.com/IBM-Cloud/vpc-tutorials/vpc-app-deploy/ansible/tf $ tf version
Terraform v0.11.14
+ provider.ibm v0.17.5

Your version of Terraform is out of date! The latest version
is 0.12.9. You can update by downloading from www.terraform.io/downloads.html
```
### Expected Behavior
1. better error message on failure
1. waiting for destroy should wait for stopped but allow the stopping state

**Comment by powellquiring** [comment]
After this failure the instance was in the stopped state, and destroy did not like that.  Allowing the stopped state might make the system more robust.
```
Error: Error refreshing state: 1 error occurred:
	* module.vpc_pub_priv.ibm_is_instance.frontend: 1 error occurred:
	* module.vpc_pub_priv.ibm_is_instance.frontend: ibm_is_instance.frontend: {"errors":[{"code":"volume_service_unavailable","message":"Service Unavailable","more_info":"http://www.bluemix.com/help#volume_service_unavailable","target":{"name":"","type":""}}],"trace":"39887a118b5984939ca3a7ecb2764c83"}
```

**Comment by powellquiring** [comment]
Added a configuration to the description.

**Comment by VaishnaviGopal** [comment]
We have identified and fixed the issue. Updates would be available in the next IBM Cloud provider release.

**Comment by albee-jhoney** [comment]
PR raised - https://github.com/IBM-Cloud/terraform-provider-ibm/pull/772

**Comment by hkantare** [comment]
Fixed in master branch.

**Comment by VaishnaviGopal** [comment]
The issue is fixed and the updates are available in IBM Cloud Provider V0.17.6. Link to download this plugin: https://github.com/IBM-Cloud/terraform-provider-ibm/releases/tag/v0.17.6

**Comment by hkantare** [comment]
closing the issue
